pkg/apis/core/types.go

```diff
@@ -3000,12 +3019,66 @@ type PodSpec struct {
 	// - spec.containers[*].securityContext.runAsGroup
 	// +optional
 	OS *PodOS
+
 	// SchedulingGates is an opaque list of values that if specified will block scheduling the pod.
 	// More info:  https://git.k8s.io/enhancements/keps/sig-scheduling/3521-pod-scheduling-readiness.
 	//
 	// This is an alpha-level feature enabled by PodSchedulingReadiness feature gate.
 	// +optional
 	SchedulingGates []PodSchedulingGate
+
+	// ResourceClaims defines which ResourceClaims must be allocated
+	// and reserved before the Pod is allowed to start. The resources
```

Comment:
That's because it is up to the driver to handle this: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3063-dynamic-resource-allocation#implementing-optional-resources

It's not part of the API. To Kubernetes, a ResourceClaim for which resources where not available will still look like it is allocated.